src/include/ipc/info.h

```diff
@@ -40,6 +40,14 @@ enum sof_ipc_ext_data {
 	SOF_IPC_EXT_WINDOW,
 };
 
+/* Compiler version */
+struct sof_ipc_cc_version {
+	uint16_t major;
+	uint16_t minor;
+	uint16_t micro;
+	const char *name;
+};
```

Comment:
needs packed. We also cant pass a char * in IPC data to host. Do we also want to include a vendor string too ? i.e. to include XCC version/config data (like used in xtensa-build-all.sh) ?

Reply:
Ok, I add attribute packed and pack data manually and change pointer to char[4] and pass there CC_NAME value. Could you write little more about vendor string? I'm not sure what you think about.

Reply:
We had some issue recently where the wrong XCC version was being used so we could include this for XCC 
```
XTENSA_TOOLS_VERSION="RG-2017.8-linux"
```
Probably not worth adding extra details from GCC atm as version is probably good enough.

Reply:
`sof_ipc_fw_ready` is declared as `const struct` in `platform.c` files, so it must be filled during compilation, no run-time `memcpy` possibility. if we would like to add this extra information as a string with max size for example of 16 chars (minus null termination) then I would use construction like this:

1)
```
#define CC_ASSIGN_DESC(field) \
	field[0] = CC_DESC[0] \
	field[1] = CC_DESC[1] \
	...
	field[1] = CC_DESC[15] \
```
But for shorter strings it show compilation error `error: initializer element is not a compile-time constant`. So we have constant length description allowed only.

2) It also may be solved by heavy preprocessor programming, like in trace module but do we like that?

3) Another option is to send enumerated value, but then after add new XCC then it will be need to update linux kernel.

Reply:
added